### ISSUE TITLE
Question about Apple Silicon and macOS support workaround 

### ISSUE THREAD

**Issue by HaskDev0** [state-spaces/mamba#130]
Hi all,

As a macOS user running on Apple Silicon I have an interest to use Mamba model on my laptop, but the Linux and NVIDIA requirements do not allow to do this. 

Does anyone know how to overcome this problem and if there is a way to train the model and use if for inference of Apple Silicone machines? E.g., rewriting some parts that utilizes Apple GPUs. 

I found that there is one attempt to use the model only for inference which uses Rust though:

https://github.com/flawedmatrix/mamba-ssm

Thanks in advance!

**Comment by albertfgu** [comment]
This is beyond our scope, but hopefully the community will step up here :)

**Comment by HaskDev0** [comment]
> This is beyond our scope, but hopefully the community will step up here :)

Thank you for commenting back! And still want to say it's a great research and work :)

**Comment by HaskDev0** [comment]
> Saw this a few days ago:
> [alxndrTL/mamba.py/mlx](https://github.com/alxndrTL/mamba.py/tree/36576b4a132ace42104506cf03e26c55383d06a3/mlx)

Thank you for the resource! I would need to check it on my device too. It's still 2x times slower though... Looks like there is a room for improvement with MLX for Apple then.

**Comment by HaskDev0** [comment]
> > Thank you for the resource! I would need to check it on my device too. It's still 2x times slower though... Looks like there is a room for improvement with MLX for Apple then.
> 
> Yeah, there are still performance cliffs in mlx, however the author of that mamba repo also notes problems with his implementation in mlx. Unfortunately, there isn't currently a competitive mamba reference implementation using mlx, but I suspect that will change soon 🙂

Yes, would be good to see this :) I might post here a link if I accidentally find something

**Comment by hg0428** [comment]
Any updates on this? Can we get Apple Silicon support?

**Comment by fil-does** [comment]
I am curious about mamba and would like to use it for research project, is there a way to use mamba with intel graphics (like iris or xe)?

**Comment by fil-does** [comment]
is there a possibility to use mamba with intel (Iris / Xe) ? i can not find any solution for it and i would love to use it in a research project.

**Comment by purohit10saurabh** [comment]
I've implemented Mamba and Mamba-2 for Apple Silicon with MPS acceleration. You can check it here: https://github.com/purohit10saurabh/mamba-ssm-macos
